srcpkgs/zathura/template

```diff
@@ -1,13 +1,13 @@
 # Template file for 'zathura'
 pkgname=zathura
 version=0.4.7
-revision=2
+revision=3
 build_style=meson
-configure_args="-Dsynctex=enabled -Dtests=disabled"
+configure_args="-Dsynctex=enabled -Dtests=enabled"
 hostmakedepends="pkg-config intltool python3-Sphinx desktop-file-utils
  appstream-glib glib-devel librsvg-utils"
 makedepends="girara-devel sqlite-devel file-devel zlib-devel libseccomp-devel
- libglib-devel texlive-devel"
+ libglib-devel texlive-devel check-devel gettext-devel"
```

Comment:
Please use `checkdepends` for these, since the package itself doesn't need them (I'd add `-Dtests=enabled/disabled` conditionally to `configure_args` as well, using `XBPS_CHECK_PKGS`).

Reply:
Ehm, if I move `check` to `checkdepends`, then I get error
```
Run-time dependency check found: NO (tried pkgconfig)
tests/meson.build:1:0: ERROR: Dependency "check" not found, tried pkgconfig
```

Reply:
is it better now?

Reply:
What command did you use that resulted in that?

Reply:
I used `./xbps-src pkg -f` on a template that had `Dtests=enabled`.

Reply:
With tests disabled, the dependencies should work when put in checkdepends.

Reply:
As soon as I have `check-devel` in `checkdeps`, it fails, even with `-Dtests=disabled`; if I remove it, it works.

Reply:
```
$ => zathura-0.4.7_3: skipping check (XBPS_CHECK_PKGS is disabled) ...
=> zathura-0.4.7_3: running pre-install hook: 00-libdir ...
=> zathura-0.4.7_3: running pre-install hook: 02-script-wrapper ...
=> zathura-0.4.7_3: running pre-install hook: 98-fixup-gir-path ...
=> zathura-0.4.7_3: running do_install ...
ninja: Entering directory `build'
ninja: error: '/usr/lib64/libcheck.so', needed by 'tests/test_document', missing and no known rule to make it
=> ERROR: zathura-0.4.7_3: do_install: 'DESTDIR=${DESTDIR} ${make_cmd} -C ${meson_builddir} ${make_install_args} ${make_install_target}' exited with 1
=> ERROR:   in do_install() at common/build-style/meson.sh:148
```

Reply:
Have you cleaned your build directory? That didn't happen previously... (unless it's gettext-devel to blame).

Reply:
Ok, I take it back, now it works if Dtests are disabled.

Reply:
Is anything like this correct then?
```
if [ "$XBPS_CHECK_PKGS" = full ]; then
	configure_args+=" -Dtests=enabled"
	makedepends+=" check-devel"
else
	configure_args+=" -Dtests=disabled"
	check-depends+=" check-devel"
fi
```

Reply:
check-devel can always be in checkdepends (just set the variable after `depends`), and should never be in makedepends.

With tests=disabled, meson should never try to look for it.

Reply:
With tests disabled, there's no problem.
If dtests are enabled, it is not enough to have it it checkdepends, it has to be in makedepends, otherwise I get error.
Do you agree?
```
if [ "$XBPS_CHECK_PKGS" = full ]; then
	configure_args+=" -Dtests=enabled"
	makedepends+=" check-devel"
fi
```

Reply:
No, `checkdepends` will be installed equivalently to how `makedepends` is, for the cases where we run tests.

```diff
diff --git a/srcpkgs/zathura/template b/srcpkgs/zathura/template
index 3a6afd35e5..31b514eb4a 100644
--- a/srcpkgs/zathura/template
+++ b/srcpkgs/zathura/template
@@ -7,8 +7,8 @@ configure_args="-Dsynctex=enabled"
 hostmakedepends="pkg-config intltool python3-Sphinx desktop-file-utils
  appstream-glib glib-devel librsvg-utils"
 makedepends="girara-devel sqlite-devel file-devel zlib-devel libseccomp-devel
- libglib-devel texlive-devel gettext-devel"
-checkdepends="check-devel"
+ libglib-devel texlive-devel"
+checkdepends="check-devel gettext-devel"
 short_desc="Highly customizable and functional document viewer"
 maintainer="lemmi <lemmi@nerd2nerd.org>"
 license="Zlib"
@@ -16,9 +16,10 @@ homepage="https://pwmt.org/projects/zathura"
 distfiles="${homepage}/download/zathura-${version}.tar.xz"
 checksum=e012dbfe2b981b826553a9af8420d42a9c5d3387fbe14a5399ce94a2d374a1e7
 
-if [ "$XBPS_CHECK_PKGS" = full ]; then
+if [ "$XBPS_CHECK_PKGS" ]; then
 	configure_args+=" -Dtests=enabled"
-	makedepends+=" check-devel"
+else
+	configure_args+=" -Dtests=disabled"
 fi
 
 post_install() {
```

Reply:
Done. Althoug it still fails on my system, if I do `pkg -f`. Perhaps it is just a local issue.